### PR TITLE
update rep 2002 with lessons learned from writing migration tool

### PR DESCRIPTION
**Pull request by tfoote**

### CODE REVIEW

File: rep-2002.rst [hunk 1]
```diff
@@ -32,14 +32,30 @@ During the pre-release phase for stable ROS distributions releases in the rollin
 Release process
 ===============
 
+Automated release migration tool
+--------------------------------
+
+To make a new rosdistro, target, from an existing rosdistro, source. An automated migration tool is available to to avoid manual labor.
```
Comment (nuclearsandwich) [inline rep-2002.rst]:
```suggestion
To make a new rosdistro, target, from an existing rosdistro, source. An automated migration tool is available to to reduce manual labor.
```

File: rep-2002.rst [hunk 2]
```diff
@@ -32,14 +32,30 @@ During the pre-release phase for stable ROS distributions releases in the rollin
...
+
+A snaphot of the source distro will be captured, most likely a git commit-ish.
+The release manager should then setup the target rosdistro with all the appropriate configurations, such as target platforms and architectures.
+The tool will then iterate through all repositories in the snapshot of the source rosdistro and attempt to automatically run bloom release on the package at the specific version in the snapshot.
+If it fails the maintainer will be needed to resolve any issues.
+The tool will continue skipping that package and all downstream repositories.
```
Comment (nuclearsandwich) [inline rep-2002.rst]: If this is meant to describe the migration script, we don't transitively skip. In fact we avoid needing to topologically sort altogether by allowing bloom to assume everything will be in the release and then removing what doesn't make it. We'll still try to bloom everything but the revoked release of a dependency will prevent the downstream jobs from being created on the buildfarm.

File: rep-2002.rst [hunk 3]
```diff
@@ -48,7 +64,10 @@ Updating the rolling release platforms
 The rolling release is intended to remain available indefinitely and so will not be retired or reach end-of-support status without an update to this REP and the ROS distribution release process.
 However, this does mean that platform changes during the life of the rolling distribution are going to occur and the rolling release may target platforms that are themselves unstable or in a pre-release state.
 When a platform change is coming, the intended date of change will be announced ahead of time.
-Once the platforms have been updated, everything currently in the rolling release will be automatically re-released via bloom to target the updated release platforms.
+
+The platform change is a unique invocation of the migration tool where the source and target rosdistro will be the same.
+To do this the person managing the migration will take the snapshot of the source rosdistro and then clear it's contents as well as updating the targetget platforms.
```
Comment (nuclearsandwich) [inline rep-2002.rst]:
```suggestion
To do this the person managing the migration will take the snapshot of the source rosdistro and then clear it's contents as well as updating the target platforms.
```

File: rep-2002.rst (hunk 2 above)
Comment (tfoote) [reply rep-2002.rst]: ok, I just pulled that form the description

**Pull request merged**